dvc/external_repo.py

```diff
@@ -59,13 +59,17 @@ def make_repo(path, **_kwargs):
         url=url,
         scm=None if for_write else Git(root_dir),
         rev=None if for_write else rev,
-        subrepos=not for_write,
-        uninitialized=True,
         config=config,
         repo_factory=make_repo,
         **kwargs,
     )
 
+    if "subrepos" not in repo_kwargs:
+        repo_kwargs["subrepos"] = True
+
+    if "uninitialized" not in repo_kwargs:
+        repo_kwargs["uninitialized"] = True
+
```

Comment:
These should be explicit as in the ls.py below. Just keeping this as a temporary compatibility layer for other code.